### DISCUSSION TITLE
Calling server action throws undefined immediately, without actually calling the server

### DISCUSSION THREAD

**Discussion by nir-peled** [vercel/next.js#67117] [Help]
### Summary

I have a form that, in `onSubmit` has a function that calls a server action function. 
When I try to submit the form, and the callback tries to call the server action, it is not called. Instead, it always throws `undefined`. 

I found someone else having the same problem [here](https://www.reddit.com/r/nextjs/comments/1dape5a/middleware_causes_server_actions_to_throw/), and a suggested solution to something similar [here](https://github.com/vercel/next.js/issues/50659), which didn't work. 
Some server actions work, like my login/logout functions using next-auth. 

Edit: Apparently, it only happens when the developer console is open. 

Minimal reproduction:
```
// in TestForm.tsx
"use client";
import React from "react";
import Button from "./Button";
import { echo } from "../lib/actions";

export default function TestForm() {
	return (
		<form onSubmit={async () => console.log(await echo("hello"))}>
			<Button type="submit">Submit</Button>
		</form>
	);
}

// in actions.ts
"use server";
// other actions 

export async function echo(param: any): Promise<any> {
	return param;
}
```

### Additional information

```js
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 11 Home
Binaries:
  Node: 20.11.0
  npm: N/A
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 14.1.4
  eslint-config-next: 14.0.4
  react: 18.3.1
  react-dom: 18.2.0
  typescript: 5.4.4
Next.js Config:
  output: N/A
```

**Comment by corwinstephen** [comment]
For anyone else who ends up here, upgrade to the latest Next, this seems to have been a bug that masks the error when you pass something invalid as input to a server action. It's fixed now.
